server/src/main/java/org/elasticsearch/index/query/QueryRewriteContext.java

```diff
@@ -407,4 +410,27 @@ public ResolvedIndices getResolvedIndices() {
     public PointInTimeBuilder getPointInTimeBuilder() {
         return pit;
     }
+
+    /**
+     * Retrieve the first tier preference from the index setting. If the setting is not
+     * present, then return null.
+     */
+    @Nullable
+    public static String getTierPreference(QueryRewriteContext context) {
+        if (context instanceof CoordinatorRewriteContext) {
+            String tier = ((CoordinatorRewriteContext) context).tier();
+            // dominant branch first (tier preference is configured)
+            return tier.isEmpty() == false ? tier : null;
+        }
+        Settings settings = context.getIndexSettings().getSettings();
+        String value = DataTier.TIER_PREFERENCE_SETTING.get(settings);
```

Comment:
We've had performance issues in the past with parsing the tier preference from the settings. We now have a version of the `IndexMetadata` that does the parsing and caches the result for the lifetime of the `IndexMetadata`. 
Since the `CoordinatorRewriteContextProvider` has access to the `IndexMetadata` I preferred to take this route rather than re-parsing the setting.